vllm/core/block/block_table.py

```diff
@@ -105,12 +108,27 @@ def append_token_ids(self,
         Args:
             token_ids (List[int]): The sequence of token IDs to be appended.
         """
-        assert self._is_allocated
+        assert self._is_allocated, "no blocks have been allocated"
         assert len(self._blocks) > 0
 
+        if self._block_sliding_window is not None:
+            null_block = self._allocator.null_block
+            if num_computed_slots is None:
+                num_computed_slots = self._num_full_slots
+            end_idx = (num_computed_slots //
+                       self._block_size) - self._block_sliding_window
+            for idx in range(0, end_idx):
+                b = self._blocks[idx]
+                if b is not null_block:
```

Comment:
during forking, is there some risk the python object equality will not hold? should we check the block number (or implement better equality)

Reply:
changed to `isinstance(last_block, NullBlock)`